### PR TITLE
Feature/file to array in main

### PR DESCRIPTION
**Pull request by DavidT12**
removed a while loop that also created a file for holding directory details. 
Added in an array that would hold the directory details
added in a for loop to look at each array element

### CODE REVIEW

File: GitUpdateScript.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ SECONDS=0  #set seconds counter back to 0
 function GoToGitRepositoryDirectory(){
 
 	cd $home_dir${file_dir:1:-4} #change to git dir
-	pwd #pint dir to screen
+	echo "Starting git repository refresh @" $(pwd) #print dir to screen
```
Comment (DavidT12) [inline GitUpdateScript.sh:16]: added echo before pwd to improve style of the output

File: GitUpdateScript.sh [hunk 2]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
 }
 
 home_dir=$(pwd)  #v1 holds the home dir
-echo "Git Repositories Found:"
```
Comment (DavidT12) [inline GitUpdateScript.sh]: echo about repositories found not relevant anymore as directories as no longer being forced to the screen when creating the file

File: GitUpdateScript.sh [hunk 3]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
...
+
 
 # todo Count from find instead of iterating loop from file
-find -name .git| tee Temp_dir_log  #finds file with name .gitignore and stores results in file called FILENAME
```
Comment (DavidT12) [inline GitUpdateScript.sh]: replaced by an array and for loop

File: GitUpdateScript.sh [hunk 4]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
...
 
-echo "Going to each git repositories and update:"
-while read file_dir
-do
 
-	GoToGitRepositoryDirectory
+arrayFileDir=($(find -name .git))
```
Comment (DavidT12) [inline GitUpdateScript.sh]: instead of creating a file to hold find results, now using an array

File: GitUpdateScript.sh [hunk 5]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
...
 	BranchChecking  #call branch checking function
 
 	git pull
 
 	echo "Elapsed Time:" $SECONDS
-done < Temp_dir_log
 
+done
 
-RemoveFileCreatedToHoldGitReposNames
```
Comment (DavidT12) [inline GitUpdateScript.sh]: file no longer created so no need to remove the file

File: GitUpdateScript.sh [hunk 6]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
 }
 
 home_dir=$(pwd)  #v1 holds the home dir
-echo "Git Repositories Found:"
+
 
 # todo Count from find instead of iterating loop from file
```
Comment (bazzani) [inline GitUpdateScript.sh]: this todo comment can be removed

File: GitUpdateScript.sh [hunk 7]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
 }
```
Comment (bazzani) [inline GitUpdateScript.sh]: you can remove the RemoveFileCreatedToHoldGitReposNames function as it is no longer called

File: GitUpdateScript.sh (hunk 4 above)
Comment (bazzani) [reply GitUpdateScript.sh]: as discussed, moving this to a function is cleaner code

File: GitUpdateScript.sh (hunk 5 above)
Comment (bazzani) [inline GitUpdateScript.sh]: Can you add a script start comment at the bottom of the file?

e.g. 
```
########
# Script start
########
```

File: GitUpdateScript.sh [hunk 8]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
...
+for (( n=0; n<$numberOfGitRepos; n++ ))
+do
+    file_dir=${arrayFileDir[$n]}
+ 	GoToGitRepositoryDirectory
 
 	BranchChecking  #call branch checking function
 
 	git pull
 
 	echo "Elapsed Time:" $SECONDS
```
Comment (bazzani) [inline GitUpdateScript.sh]: add an echo under this line to separate the logging per git repo

File: GitUpdateScript.sh (hunk 1 above)
Comment (bazzani) [reply GitUpdateScript.sh:16]: +1

**Review by bazzani — CHANGES_REQUESTED** [review]

File: GitUpdateScript.sh [hunk 9]
```diff
@@ -94,26 +94,30 @@ rm Temp_dir_log #rm removes/dels the file
...
-do
 
-	GoToGitRepositoryDirectory
+arrayFileDir=($(find -name .git))
+numberOfGitRepos=$(find -name .git|wc -l)  #wc is word count -l makes it count lines, | is pipe and find finds all .git
+
+
+echo "Going to each git repository and updating:"
+
+for (( n=0; n<$numberOfGitRepos; n++ ))
```
Comment (bazzani) [inline GitUpdateScript.sh]: do you need double parenthesis here? ((

File: GitUpdateScript.sh [hunk 10]
```diff
@@ -13,7 +13,7 @@ SECONDS=0  #set seconds counter back to 0
 function GoToGitRepositoryDirectory(){
 
 	cd $home_dir${file_dir:1:-4} #change to git dir
-	pwd #pint dir to screen
+	echo "Starting git repository refresh @" $(pwd) #print dir to screen
 }
 function FindsWhatBranchRepositoryIsOn() {
 	if [ "$branch_option" = "* $branch_I_want" ] #note "* " is used to show branch selection
```
Comment (v1-robert-metcalfe) [inline GitUpdateScript.sh]: I'd recommend using `[[` over `[`. It's not POSIX compliant, but works in `bash`, `ksh`, and `zsh`, and is more feature-rich.

[For your reading](https://unix.stackexchange.com/a/32211/67145).

**Review by v1-robert-metcalfe — COMMENTED** [review]
For all scripts you write, I'd recommend you lint them. Try https://www.shellcheck.net/.

File: GitUpdateScript.sh (hunk 9 above)
Comment (v1-robert-metcalfe) [reply GitUpdateScript.sh]: Double brackets are for arithmetic operations. You can omit dollar signs from your variables for readability. `$n++` becomes `n++`.

File: GitUpdateScript.sh (hunk 10 above)
Comment (DavidT12) [reply GitUpdateScript.sh]: Thanks will do

File: GitUpdateScript.sh [hunk 11]
```diff
@@ -13,46 +13,46 @@ SECONDS=0  #set seconds counter back to 0
 function GoToGitRepositoryDirectory(){
 
 	cd $home_dir${file_dir:1:-4} #change to git dir
-	pwd #pint dir to screen
+	echo "Starting git repository refresh @" $(pwd) #print dir to screen
 }
 function FindsWhatBranchRepositoryIsOn() {
-	if [ "$branch_option" = "* $branch_I_want" ] #note "* " is used to show branch selection
+	if [[ "$branch_option" = "* $branch_I_want" ]] #note "* " is used to show branch selection
```
Comment (DavidT12) [inline GitUpdateScript.sh:19]: notation change from [ to [[

File: GitUpdateScript.sh [hunk 12]
```diff
@@ -87,33 +87,35 @@ function BranchChecking(){
 	MoveToCorrectBranch
 
 }
+function FindGitReposotories()
```
Comment (DavidT12) [inline GitUpdateScript.sh:90]: function added to clean up the code

File: GitUpdateScript.sh [hunk 13]
```diff
@@ -87,33 +87,34 @@ function BranchChecking(){
...
-home_dir=$(pwd)  #v1 holds the home dir
-echo "Git Repositories Found:"
-
-# todo Count from find instead of iterating loop from file
-find -name .git| tee Temp_dir_log  #finds file with name .gitignore and stores results in file called FILENAME
+    echo "Going to each git repository and updating:"
 
-echo "Going to each git repositories and update:"
-while read file_dir
+for (( n=0; n<$numberOfGitRepos; n++ ))
```
Comment (bazzani) [inline GitUpdateScript.sh]: can you indent the for loop?

File: GitUpdateScript.sh [hunk 14]
```diff
@@ -87,33 +87,34 @@ function BranchChecking(){
...
 	BranchChecking  #call branch checking function
 
 	git pull
 
 	echo "Elapsed Time:" $SECONDS
-done < Temp_dir_log
+    echo ""
+done
+}
+#########################################
```
Comment (bazzani) [inline GitUpdateScript.sh]: 2 x lines of white space please

**Pull request merged**